### DISCUSSION TITLE
Resource crates and markets

### DISCUSSION THREAD

**Discussion by 12gabriel3** [EveronLife/EveronLife#146] [Ideas]
# Resource crates and markets

Resource crates can be a good addition to the resource and inventory system. While markets would integrate very well into the player driven economy.

## Current situation

In the Arma 3 implementation of life servers, when you gather or process, the items spawn in your virtual inventory. You sell them at fixed locations on the map for each item.

## New system

With the crates system, you would take a crate from the vendor, fill it with goods and store it in your truck. You can't transport resources in your inventory. This solves many of the rough edges of the current system. It simplifies the production, while getting rid of the virtual inventory.

### Gathering with crates

To gather with crates you would need to go to the gathering point, take a crate from the vendor and fill it with resources. This last step could be done in multiple ways. In the case of fruits, you could take the crate and start collecting. In the case of minerals, you could break rocks with a pickaxe, place the crate near you and fill it with a shovel. In case of wood, you could chop the tree down, saw the tree trunk into smaller pieces and place them in the crate.

### Loading the crate

To load the crate into the truck, just hold the crate and interact with the trunk. Each vehicle can hold a specific number of crates.

### Processing with crates

To process the items you collected, interact with the trunk to take a crate, then, with the crate in hand, interact with the processor to dump the crate contents and then turn it on. The output is in the form of crates as well.

### Selling crates

Go to the vendor with the crate in your hand and interact to sell the crate.

## Economic system

We have agreed to make a player driven economy. We could implement that in the form of _resource pools_. Instead of having one place to sell the items, we could have multiple locations. Each location is a _distribution center_ (_DC_) that buys and sells any type of crate. The crates feed a pool of resources when sold and drain when bought. The pool is also drained by stores whenever a player buys an item. The price for each _pool item_ is set dynamically, according to supply and demand, for each _DC_.

### Distribution centers

Located in the main cities, they are a central market for crates. They are a safe spot for players to engage in economic activities. A player can:

* Sell unprocessed or processed crates for profit
* Buy unprocessed crates, process them and sell the processed crate for profit
* Buy a type of crate from a DC where it is cheap and sell it to another DC for profit

Each crate feeds the pool of the corresponding item. They are not physical items, but "virtual", much like supplies in conflict mode. They can only be transported with crates, cannot be consumed, but can be processed, bought and sold. They could also have other uses, but more on that later.

### "Virtual" items

Virtual items in this system are not the same as virtual items in Arma 3 life. Virtual items cannot be interacted with directly, meaning they don't take space in the inventory. They can be interacted with indirectly with crates or by buying things on stores. Virtual items could include:

* Weapon parts
* Vehicle parts
* Gas
* Food
* Drink
* Different drugs

Each crate holds a specific amount of virtual items. It could be set globally or for each specific crate. If a crate holds 100 virtual items, selling it would increase the pool by 100 while buying it would drain 100 from the pool.

### Stores

Items sold on stores are physical:

* Food
* Weapons
* Clothes
* Compasses
* Vehicles
* Drugs

Or "virtual":

* Gas

Each item drains from a pool of virtual items:

* M16 is 100 weapon parts
* Meal is 50 fish filet and 50 fruit
* M151A2 (jeep) is 100 car parts
* Weed joint is 100 marijuana

### Demand

Every "virtual" item should have demand. Drugs can be consumables that give bonuses, weapons/kits should be lost on death, cars can be lost/destroyed/require maintenance. Here are some consumable effects of drugs I could think of:

#### Marijuana

Increases max health

#### Meth

Increases driving speed.

#### Cocaine

Increases stamina.

#### Beer

Decreases _buy_ price of crates (not _sell_ price). This is done to encourage players to drive drunk. After they buy goods, they transport them in a drunken state.

#### Liquor

Same thought process as _beer_, but it increases processing yield.

Liquor and beer could both drain from the same _Alcohol_ pool.

### Contraband

The player could choose wether they do the _paperwork_ for each crate on gather/processing as an extra step, making the run take more time to complete, or take the riskier route of doing _contraband_. This means any item can be transported as _contraband_.

### Dynamic yield

Each spot could have a different yield depending if the crates produced there are reaching the proper destination (not being robbed, destroyed or apprehended). There could be a place on the map that accepts stolen goods. It could be a port that pays less than DCs (to discourage players from stealing goods to increase yield).
Using a place too much should decrease yield.

### Personal pools

Crates could also feed into "personal" pools. This could be use by gangs, players with houses and police to feed their armory or other things such as a garage to get access to items directly without having to go to a store and buy it directly.

**Comment by Arkensor** [comment]
I am sorry but I do not really understand the benefit of putting everything into a crate. How is putting gathered fruits into a crate a better option than just having a bunch of them in your backpack? 

If I understand correctly you want any resources gathered to be inside crates? So all items look the same except it says "Box of xyz". I do not think that is a pleasant experience.

All the player-driven economy aspects can be achieved without putting things into boxes. Yes altis life was mainly a linear route, but that was due to lack of creativity. With things like sand right now it can be used to produce glass bottles that could be sold directly or sold to other players for more money, so they can fill apple juice into them. Intermediate resources could just be bought for a lot, players can sell for half of that and so it makes sense to sell to other players for 150% because they would otherwise pay 200% to buy it from the NPC.

Crates can be used to manipulate larger quantities of something as a "helpful organization tool" but I do not yet see a reason why they would become the main gameplay. Can you please try to summarize why it needs boxes in like 2 or 3 sentences?

**Reply by 12gabriel3** [reply]
Firstly, each container can be vastly different from one another. Examples:
![image](https://user-images.githubusercontent.com/16885097/182042057-02054b81-b10b-4b38-8e5e-c6a79de8b943.png)![image](https://user-images.githubusercontent.com/16885097/182042064-74191bb5-c992-43f2-bcae-b0490c3dab28.png)
![image](https://user-images.githubusercontent.com/16885097/182042131-f04f5636-af84-4280-b18f-f7ddc9d76a2c.png)

![image](https://user-images.githubusercontent.com/16885097/182042109-14d42d5a-be7d-4f41-bd83-4004c2cc1ae7.png)


A dynamic economy doesn't really _need_ boxes. Let me make explain my idea better.
I really dislike the inventory system in Arma 3 Life. One thing is the player carry capacity, which is too much, more than most cars, which doesn't make sense. Another thing is putting weird things into your backpack, such as ore or sand. This is done because the player needs a way to transport the items from place to place.
I wanted to come up with a system that does 2 things:
* Allows the player to carry more weight and volume than usual by picking up things with both hands (crates/barrels/logs or whatever)
* Makes the vehicle the central part of transport, instead of the backpack. You don't see truckers transporting cargo in their backpacks IRL. That should be made either illegal (against transport rules) or impossible (too much weight/volume).

The crate system, or we could call it the "minimum commercial quantity system", would solve that. If we make the minimum quantity that can be sold larger than the players backpack and constrain certain items to a certain container, such as sand to sacks, it would solve both issues.
Putting items inside containers as "virtual" items would simplify the development of some interesting features, I think. Example: hiding illegal items in double bottom crates or inside sand bags.
The _paperwork_ step would also be easier and more intuitive with crates, imo. Making every item possibly illegal is a feature that I think would add a lot to the game mode.
The dynamic pricing system would work best with fixed quantities that can be sold a time. Let me exemplify:
Apples are bought for 10 Everos. Selling apples should make the prices drop. Lets say that selling 1000 apples makes the price drop to 8 Everos. When a player comes and sell 1000 apples we have 2 options:
1. The price drops for each apple sold. The player will get a mean price of 9 Everos. They will feel cheated.
2. The price drops only _after_ all apples are sold. This means that the bigger their inventory space, the best their price. So for smaller items that can be sold all at once, you will get a better price

By having a fixed quantity that can be sold at a time, you can do scenario 2. But this time, for each box sold, the price is updated. The player knows exactly how much he will earn for each box, and inventory space doesn't matter because they can only sell one _commercial unit_ at a time. They can decide to sell part of the cargo in one place, and when the price hits a certain threshold, sell the rest at another DC

Overall I think this system fits very well with the whole economy system, even though it is not strictly necessary.

**Reply by thebonbon** [reply]
A dynamic economy system with different traders would make things less static and you accually have to plan your farming routes.

> I really dislike the inventory system in Arma 3 Life. One thing is the player carry capacity, which is too much, more than most cars, which doesn't make sense. Another thing is putting weird things into your backpack, such as ore or sand. This is done because the player needs a way to transport the items from place to place.
>
For sand and stuff we have bags. Yeah its kinda weird to put a bunch of sandbags in your backpack. 
Thats why I tried another route with my logging system, that you need to pickup the big logs and carry them to a car.
For really heavy stuff like sandbags we can use the "on the shoulder carry" / carry in crates and make the items heavy so they dont fit in a backpack. For items like Oil barrels this would work great. (Buy empty barrels, fill them, carry them to the car)
But for the majority of item we should keep it simple and just use inventory items.

> Selling apples should make the prices drop. 

For the price drops why not make it every 30min - 60min and based on every sale in that period? 

> Makes the vehicle the central part of transport, instead of the backpack. You don't see truckers transporting cargo in their backpacks IRL. That should be made either illegal (against transport rules) or impossible (too much weight/volume).
>
I mean they could if they transported apples or something small. Idk why they would do that but possible right? Should not be illegal.

> Crates can be used to manipulate larger quantities of something as a "helpful organization tool" 
>
I also think crates should only be there as a "helpful organization tool" (Only for some items that need crates to carry). So you dont need to run from your car to the trader 100 times, just buy a crate fill it up and only run 5 times back and forth.

**Comment by jasper-one** [comment]
The title seems misleading to me, this seems like a jumble of ~10 different ideas, many of which are not related (e.g. Drug Effects).

I don't really want to spend any time critiquing the idea/s until they are a little better organized. The virtual items concept for example is something I'm very interested in staying very, very far away from.
